crates/ra_ide/src/completion/complete_unqualified_path.rs

```diff
@@ -31,6 +56,57 @@ pub(super) fn complete_unqualified_path(acc: &mut Completions, ctx: &CompletionC
     });
 }
 
+macro_rules! impl_patternlike {
+    ($($t : ty),+) => {
+        $(
+            impl PatternLike for $t {
+                fn pat(&self) -> Option<ast::Pat> {
+                    self.pat()
+                }
+            }
+        )+
+    };
+}
+
+trait PatternLike: AstNode {
+    fn pat(&self) -> Option<ast::Pat>;
+}
+
+impl_patternlike!(
+    ast::ForExpr,
+    ast::MatchArm,
+    ast::ParenPat,
+    ast::RefPat,
+    ast::BindPat,
+    ast::BoxPat,
+    ast::RecordFieldPat,
+    ast::LetStmt,
+    ast::Condition,
+    ast::Param
+);
+
+fn complete_pat_enum_variants<C: PatternLike>(
+    node: &SyntaxNode,
+    acc: &mut Completions,
+    ctx: &CompletionContext,
+) -> Option<()> {
+    if let Some(pat) = C::cast(node.clone()).and_then(|a| a.pat()) {
+        if let Some(ty) = ctx.sema.type_of_pat(&pat) {
+            if let Some(Adt::Enum(enum_data)) = ty.as_adt() {
+                let variants = enum_data.variants(ctx.db);
+                for variant in variants {
+                    let mut local_name = enum_data.name(ctx.db).to_string();
```

Comment:
we can use `find_use_path` to instead build a proper path to the enum that will take reexports etc. into account.